kolibri/utils/server.py

```diff
@@ -297,13 +297,21 @@ def STOP(self):
         unregister_zeroconf_service()
 
     def run(self):
-        from kolibri.core.discovery.utils.network.search import ZEROCONF_STATE
         from kolibri.core.discovery.utils.network.search import (
-            reinitialize_zeroconf_if_network_has_changed,
+            ZEROCONF_STATE,
+            register_zeroconf_service,
         )
 
-        if ZEROCONF_STATE["service"] is not None:
-            reinitialize_zeroconf_if_network_has_changed()
+        if (
+            ZEROCONF_STATE["addresses"] == set(get_all_addresses())
+            or ZEROCONF_STATE["port"] is None
```

Comment:
What's the condition under which we would have reached this point and the port would still be `None`? If the server has been stopped? Or hasn't started yet? (either a slightly more semantically obvious condition, or a comment, could be good)

Reply:
I'll add a comment - yeah just if the initial registration of zeroconf hasn't happened yet.